### PR TITLE
moved APFSTime to PosixTimeInNanoseconds

### PR DESCRIPTION
**Pull request by aguilajesus**
Part of #116. As suggested in PR #119 I moved integer boundaries checks to APFSTime()

### CODE REVIEW

File: dfdatetime/apfs_time.py [hunk 1]
```diff
@@ -5,37 +5,19 @@
 
 import decimal
 
-from dfdatetime import definitions
-from dfdatetime import interface
 from dfdatetime import posix_time
+from dfdatetime import definitions
```
Comment (joachimmetz) [inline dfdatetime/apfs_time.py]: Please use alphabetic ordering

File: dfdatetime/apfs_time.py [hunk 2]
```diff
@@ -67,31 +49,12 @@ def CopyFromDateTimeString(self, time_string):
...
-    timestamp *= definitions.NANOSECONDS_PER_SECOND
-
-    if microseconds:
-      nanoseconds = microseconds * definitions.MILLISECONDS_PER_SECOND
-      timestamp += nanoseconds
+    self._CopyFromDateTimeString(time_string)
 
     # Maximum value for APFS time is 2262-04-11 16:47:16.854775807
-    if timestamp > self._INT64_MAX:
+    if self._timestamp > self._INT64_MAX:
```
Comment (joachimmetz) [inline dfdatetime/apfs_time.py]: Please check lower bound as well

File: dfdatetime/apfs_time.py [hunk 3]
```diff
@@ -67,31 +49,12 @@ def CopyFromDateTimeString(self, time_string):
...
-    microseconds = date_time_values.get('microseconds', None)
-
-    timestamp = self._GetNumberOfSecondsFromElements(
-        year, month, day_of_month, hours, minutes, seconds)
-    timestamp *= definitions.NANOSECONDS_PER_SECOND
-
-    if microseconds:
-      nanoseconds = microseconds * definitions.MILLISECONDS_PER_SECOND
-      timestamp += nanoseconds
+    self._CopyFromDateTimeString(time_string)
```
Comment (joachimmetz) [inline dfdatetime/apfs_time.py]: Use `super(APFSTime, self)._CopyFromDateTimeString()`

File: dfdatetime/apfs_time.py [hunk 4]
```diff
@@ -103,12 +66,4 @@ def CopyToDateTimeString(self):
...
-    timestamp, nanoseconds = divmod(
-        self._timestamp, definitions.NANOSECONDS_PER_SECOND)
-    number_of_days, hours, minutes, seconds = self._GetTimeValues(timestamp)
-
-    year, month, day_of_month = self._GetDateValuesWithEpoch(
-        number_of_days, self._EPOCH)
-
-    return '{0:04d}-{1:02d}-{2:02d} {3:02d}:{4:02d}:{5:02d}.{6:09d}'.format(
-        year, month, day_of_month, hours, minutes, seconds, nanoseconds)
+    return self._CopyToDateTimeString()
```
Comment (joachimmetz) [inline dfdatetime/apfs_time.py]: Use super

**Comment by joachimmetz** [comment]
sry for the delay, finally got around to it. Some small nits

**Comment by aguilajesus** [comment]
Thanks for the review, @joachimmetz 
PTAL

**Pull request merged**